quarkchain/cluster/jsonrpc.py

```diff
@@ -1428,5 +1463,48 @@ def shutdown(self):
         return "pong"
 
     @public_methods.add
-    async def echo(self, params):
-        return params
+    async def echo(self, params, context):
+        if context is not None:
+            response = {"jsonrpc": "2.0", "result": params, "id": 1}
+            await context.send(str(response))
+        # return params
+
+    @public_methods.add
+    async def subscribe(self, sub_type, full_shard_id, context):
+        if context is None or full_shard_id is None:
+            return None
+        sub_id = os.urandom(16).hex()
+        response = {"jsonrpc": "2.0", "result": sub_id, "id": 1}  # TODO: change id
+        await context.send(str(response))
+        if sub_type == "newHeads":
+            await self.fetch_new_head(sub_id, full_shard_id, context)
+        else:
+            print("other types of subscription")
+            self.subscribers[sub_type].append(sub_id)
+        # return str(sub_id)
+
+    async def fetch_new_head(self, sub_id, full_shard_id, context):
+        def response_transcoder(header_info):
+
+            return {
+                "jsonrpc": "2.0",
+                "method": "eth_subscription",
+                "result": {**header_info},
+                "subscription": sub_id,
+            }
+
+        full_shard_id = shard_id_decoder(full_shard_id)
+        branch = Branch(full_shard_id)
+        shard = self.slave.shards.get(branch, None)
```

Comment:
Maybe `shard` can be moved to `subscribe()` for the use of other future types?
`response_transcoder ()` as well.